### PR TITLE
fix: console.error instead of console.log

### LINKED ISSUE #27: Use console.error in place of console.log to report errors
We seems to use console.log both for errors and other cases.

We should investigate and update to use console.error in the right places.

### PR DESCRIPTION
**Pull request by Eomm**
Fixes #27 

Opening draft to get some hints to get the `stderr` output since this configuration throws:

```
TypeError [ERR_INVALID_OPT_VALUE]: The value "Writable {
....cut...
}" is invalid for option "stdio"
```

### CODE REVIEW

**Review by rodion-arr — COMMENTED** [review]
I assume created `Writable` stream is not attached to any descriptor as required according to docs:
https://nodejs.org/api/child_process.html#child_process_options_stdio
> 5. `<Stream>` object: Share a readable or writable stream **that refers to a tty, file, socket, or a pipe with the child process**. The stream's underlying file descriptor is duplicated in the child process to the fd that corresponds to the index in the stdio array. The stream must have an underlying descriptor (file streams do not until the 'open' event has occurred).

File: .gitignore [hunk 1]
```diff
@@ -88,4 +88,7 @@ typings/
 .fusebox/
 
 # DynamoDB Local files
-.dynamodb/
\ No newline at end of file
+.dynamodb/
+
+# IDE VSCode settings files
+.vscode/
```
Comment (ljharb) [inline .gitignore]: Specific ides should go in your global gitignore, rather than in every project you touch.

File: test/index.js [hunk 2]
```diff
@@ -45,18 +46,42 @@ const assert = require('assert')
   const cliTestsDir = path.join(__dirname, 'cli');
   const cliTests = await fs.readdir(cliTestsDir);
   suite('cli tests', async () => {
-    const supportCommand = '../../../bin/support';
-    cliTests.forEach((testDir) => {
+    const supportCommand = path.normalize('../../../bin/support');
+    cliTests.forEach((testDir, i) => {
```
Comment (ljharb) [inline test/index.js]:
```suggestion
    cliTests.forEach((testDir) => {
```

File: test/index.js [hunk 3]
```diff
@@ -45,18 +46,42 @@ const assert = require('assert')
...
-        const result = childProcess.execSync(`${supportCommand} ${command}`, { cwd: cwd }).toString();
+        // const expectedErr = fs.readFileSync(path.join(cwd, 'expected-errors')).toString(); // TODO
+        const expectedErr = 'TODO';
+
+        const stdOut = buildStringWriter({ objectMode: false });
+        const processOpts = {
+          cwd: cwd,
+          stdio: [0, 'pipe', stdOut]
+        };
+        const result = childProcess.execSync(`${supportCommand} ${command}`, processOpts).toString();
```
Comment (ljharb) [inline test/index.js]:
```suggestion
        const result = String(childProcess.execSync(`${supportCommand} ${command}`, processOpts));
```

File: test/index.js [hunk 4]
```diff
@@ -45,18 +46,42 @@ const assert = require('assert')
...
 
   run();
 })();
+
+function buildStringWriter (opts) {
+  const data = [];
+  const stream = new Writable({
+    ...opts,
+    write (chunk, encoding, done) {
+      data.push(chunk.toString());
```
Comment (ljharb) [inline test/index.js]:
```suggestion
      data.push(String(chunk));
```

**Review by ljharb — COMMENTED** [review]
sorry i can’t help with the stream part.

File: test/index.js (hunk 3 above)
Comment (rodion-arr) [reply test/index.js]: @ljharb Could you please elaborate a bit more here with this string constructor suggestion? Is it some kind of ergonomic thing or there is something else?

Comment (ljharb) [reply test/index.js]: Calling .toString directly isn’t guaranteed to return a string; it’s better to always use `String()`.

File: test/index.js [hunk 5]
```diff
@@ -45,18 +45,34 @@ const assert = require('assert')
...
       });
     });
   });
 
   run();
 })();
+
+function safeReadFile (filePath) {
+  try {
+    return fs.readFileSync(filePath).toString();
```
Comment (ljharb) [inline test/index.js]: nit:
```suggestion
    return String(fs.readFileSync(filePath));
```

**Review by ljharb — APPROVED** [review]

**Review by rodion-arr — APPROVED** [review]
👍🏻

**Review by mhdawson — APPROVED** [review]
LGTM

**Pull request merged**